apps/src/templates/manageStudents/ManageCodeReviewGroups.jsx

```diff
@@ -1,31 +1,139 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import PropTypes from 'prop-types';
 import Button from '@cdo/apps/templates/Button';
+import color from '@cdo/apps/util/color';
 import i18n from '@cdo/locale';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 import StylizedBaseDialog from '@cdo/apps/componentLibrary/StylizedBaseDialog';
-import CodeReviewGroupsLoader from '@cdo/apps/templates/codeReviewGroups/CodeReviewGroupsLoader';
 import CodeReviewGroupsStatusToggle from '../codeReviewGroups/CodeReviewGroupsStatusToggle';
+import CodeReviewGroupsManager from '@cdo/apps/templates/codeReviewGroups/CodeReviewGroupsManager';
 
 const DIALOG_WIDTH = 1000;
+const STATUS_MESSAGE_TIME_MS = 5000;
+
+const SUBMIT_STATES = {
+  DEFAULT: 'default',
+  SUBMITTING: 'submitting',
+  SUCCESS: 'success',
+  ERROR: 'error'
+};
+
+const LOADING_STATES = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  ERROR: 'error'
+};
 
 export default function ManageCodeReviewGroups({
   buttonContainerStyle,
-  sectionId
+  dataApi
 }) {
+  const [groups, setGroups] = useState([]);
+  const [groupsHaveChanged, setGroupsHaveChanged] = useState(false);
+  const [submitStatus, setSubmitStatus] = useState(SUBMIT_STATES.DEFAULT);
+  const [loadingStatus, setLoadingStatus] = useState(LOADING_STATES.LOADING);
   const [isDialogOpen, setIsDialogOpen] = useState(false);
 
   const openDialog = () => setIsDialogOpen(true);
   const onDialogClose = () => setIsDialogOpen(false);
 
+  const setInitialGroups = groups => setGroups(groups);
+  const setGroupsWrapper = groups => {
+    setGroupsHaveChanged(true);
+    setGroups(groups);
+  };
```

Comment:
nit: in order to not need this wrapper, I think you could potentially add a `useEffect` hook that depends only on the `groups` value? In theory that should only fire when groups have been so you can call `setGroupsHaveChanged` there?

Reply:
mmm good idea! will give it a shot

Reply:
I think to do that I'd have to keep track of whether groups have been initialized (because I don't want this to be called when groups is changed from [] to the initial set of groups on load). Given that, I'm not sure if that's easier than what I have here?

Reply:
ahh okay gotcha. Maybe more specific name might help then? `setGroupsAndMarkChanged` or something?

Reply:
Renamed to `onGroupsUpdate` b/c now includes a third piece of state management per some feedback from Molly, so naming gets pretty wordy.